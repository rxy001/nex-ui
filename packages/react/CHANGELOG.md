# @nex-ui/react

## 0.0.15

### Patch Changes

- fdd4483: Release
- Updated dependencies [fdd4483]
  - @nex-ui/styled@0.0.15
  - @nex-ui/system@0.0.15
  - @nex-ui/hooks@0.0.15
  - @nex-ui/icons@0.0.15
  - @nex-ui/utils@0.0.15

## 0.0.14

### Patch Changes

- ca25983: Release
- Updated dependencies [ca25983]
  - @nex-ui/styled@0.0.14
  - @nex-ui/system@0.0.14
  - @nex-ui/hooks@0.0.14
  - @nex-ui/icons@0.0.14
  - @nex-ui/utils@0.0.14

## 0.0.13

### Patch Changes

- 1b97788: Release
- Updated dependencies [1b97788]
  - @nex-ui/styled@0.0.13
  - @nex-ui/system@0.0.13
  - @nex-ui/hooks@0.0.13
  - @nex-ui/icons@0.0.13
  - @nex-ui/utils@0.0.13

## 0.0.12

### Patch Changes

- 664eb21: Test

## 0.0.11

### Patch Changes

- 259356e: Test
- Updated dependencies [259356e]
  - @nex-ui/styled@0.0.11
  - @nex-ui/system@0.0.11
  - @nex-ui/hooks@0.0.11
  - @nex-ui/icons@0.0.11
  - @nex-ui/utils@0.0.10

## 0.0.10

### Patch Changes

- 5a229bd: release
- Updated dependencies [5a229bd]
  - @nex-ui/styled@0.0.10
  - @nex-ui/system@0.0.10
  - @nex-ui/icons@0.0.10

## 0.0.9

### Patch Changes

- 4a5c359: release
- Updated dependencies [4a5c359]
  - @nex-ui/styled@0.0.9
  - @nex-ui/system@0.0.9
  - @nex-ui/hooks@0.0.9
  - @nex-ui/icons@0.0.9
  - @nex-ui/utils@0.0.9

## 0.0.8

### Patch Changes

- c8c1956: release
- Updated dependencies [c8c1956]
  - @nex-ui/styled@0.0.8
  - @nex-ui/system@0.0.8
  - @nex-ui/hooks@0.0.8
  - @nex-ui/icons@0.0.8
  - @nex-ui/utils@0.0.8

## 0.0.7

### Patch Changes

- 2618e1a: release
- Updated dependencies [2618e1a]
  - @nex-ui/styled@0.0.7
  - @nex-ui/system@0.0.7
  - @nex-ui/hooks@0.0.7
  - @nex-ui/icons@0.0.7
  - @nex-ui/utils@0.0.7

## 0.0.6

### Patch Changes

- 87191c1: release
- Updated dependencies [87191c1]
  - @nex-ui/styled@0.0.6
  - @nex-ui/system@0.0.6
  - @nex-ui/hooks@0.0.6
  - @nex-ui/icons@0.0.6
  - @nex-ui/utils@0.0.6

## 0.0.5

### Patch Changes

- 7631577: release
- Updated dependencies [7631577]
  - @nex-ui/styled@0.0.5
  - @nex-ui/system@0.0.5
  - @nex-ui/hooks@0.0.5
  - @nex-ui/icons@0.0.5
  - @nex-ui/utils@0.0.5

## 0.0.4

### Patch Changes

- 70cb478: v0.0.4
- Updated dependencies [70cb478]
  - @nex-ui/styled@0.0.4
  - @nex-ui/system@0.0.4
  - @nex-ui/hooks@0.0.4
  - @nex-ui/icons@0.0.4
  - @nex-ui/utils@0.0.4
