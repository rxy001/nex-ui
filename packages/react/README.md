<div style="background:white;">
  <p align="center">
    <img src='../../assets/logo.png' style="width: 150px; height: 150px;" />
  </p>
  <h1 align="center">Nex UI</h1>
  <p align="center">ðŸŽ‰ A beautiful, modern, and reliable React component library.</p>
  <p align="center"><b>Theme Customizable, Light & Dark UI, Unique DX, Written In TypeScript</b></p>
</div>

## Documentation

Visit https://nex-ui-docs.vercel.app to view the full documentation.

## Getting Started

Visit https://nex-ui-docs.vercel.app/docs/getting-started to get started with Nex UI.

## Storybook

Visit https://nex-ui-storybook.vercel.app to view the storybook for all components.

## Contributing

Contributions are always welcome!

See [CONTRIBUTING.md](/CONTRIBUTING.md) for ways to get started.

Please adhere to this project's [CODE_OF_CONDUCT](/CODE_OF_CONDUCT.md).

## License

[MIT](https://choosealicense.com/licenses/mit/)
