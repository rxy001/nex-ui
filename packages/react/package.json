{
  "name": "@nex-ui/react",
  "version": "0.0.12",
  "author": "X1ng Yu",
  "description": "ðŸŽ‰ A beautiful, modern, and reliable React component library.",
  "homepage": "https://nex-ui-docs.vercel.app/",
  "private": false,
  "keywords": [
    "react",
    "react-component",
    "nui",
    "nex-ui"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/rxy001/nex-ui.git",
    "directory": "packages/react"
  },
  "license": "MIT",
  "scripts": {
    "build": "tsx ../../scripts/build.ts --dts",
    "build:fast": "tsx ../../scripts/build.ts",
    "typecheck": "tsc --noEmit",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "type": "module",
  "files": [
    "dist"
  ],
  "main": "src/index.ts",
  "devDependencies": {
    "@storybook/react": "8.6.12"
  },
  "dependencies": {
    "@nex-ui/hooks": "workspace:*",
    "@nex-ui/icons": "workspace:*",
    "@nex-ui/styled": "workspace:*",
    "@nex-ui/system": "workspace:*",
    "@nex-ui/utils": "workspace:*",
    "clsx": "2.1.1"
  },
  "peerDependencies": {
    "@emotion/react": ">=11",
    "motion": "^12.7.2",
    "react": ">=19",
    "react-dom": ">=19"
  },
  "sideEffects": false,
  "clean-package": "../../clean-package.config.json"
}
