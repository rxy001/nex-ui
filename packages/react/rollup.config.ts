import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { defineConfig } from 'rollup'
import typescript from '@rollup/plugin-typescript'
import del from 'rollup-plugin-delete'
import dts from 'rollup-plugin-dts'
import { nexUIRollupPlugin } from '@nex-ui/plugins'

const dirname = fileURLToPath(new URL('./', import.meta.url))

export default defineConfig([
  {
    plugins: [
      del({
        targets: './dist/*',
      }),
      typescript({
        tsconfig: path.resolve(dirname, 'tsconfig.json'),
      }),
      nexUIRollupPlugin(),
    ],
    input: './src/index.ts',
    external: [
      'react',
      'react-dom',
      'react/jsx-runtime',
      'react-dom/client',
      '@nex-ui/utils',
      '@nex-ui/css-system',
      'classnames',
      'framer-motion',
      '@iconify/react',
      '@vanilla-extract/dynamic',
      '@vanilla-extract/css',
      '@vanilla-extract/css/functionSerializer',
    ],
    output: [
      {
        preserveModules: true,
        format: 'es',
        dir: './dist/es',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
      {
        preserveModules: true,
        format: 'cjs',
        dir: './dist/lib',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
    ],
  },
  {
    plugins: [dts()],
    input: './src/index.ts',
    output: {
      dir: './dist/types',
      format: 'es',
      preserveModules: true,
    },
  },
])
