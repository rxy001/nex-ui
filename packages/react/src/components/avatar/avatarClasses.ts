import { generateUtilityClasses } from '../utils'

export const avatarClasses = generateUtilityClasses('nui-avatar', [
  'root',
  // "sm" | "md" | "lg" | "full"
  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-full',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'img',
])
