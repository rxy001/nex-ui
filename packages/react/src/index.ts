import './types/declare'

export * from './components'
export * from './theme'

export type { Theme } from './types/theme'
export type { ComponentsTheme } from './types/componentsTheme'

export { useColorScheme } from '@nex-ui/system'
export { nex, styled } from '@nex-ui/styled'
