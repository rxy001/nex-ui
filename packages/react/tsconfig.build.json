{
  "extends": "../../tsconfig.json",
  "include": ["./src"],
  "exclude": ["./src/**/__tests__/**/*", "./src/**/__stories__/**/*"]
}
