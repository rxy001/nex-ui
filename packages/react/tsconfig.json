{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "~/*": ["../../*"],
      "@theme": ["./src/theme"]
    }
  },
  "include": ["./src", "./rollup.config.ts"]
}
