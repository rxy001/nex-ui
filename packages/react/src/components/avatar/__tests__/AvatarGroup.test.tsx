import { createRef } from 'react'
import {
  testComponentStability,
  renderWithNexUIProvider,
  testRootClassName,
} from '~/tests/shared'
import { AvatarGroup, Avatar } from '../index'
import { avatarClasses, avatarGroupClasses } from '../avatarClasses'

describe('AvatarGroup', () => {
  testComponentStability(
    <AvatarGroup>
      <Avatar />
      <Avatar />
      <Avatar />
    </AvatarGroup>,
  )

  testRootClassName(
    <AvatarGroup className='test-class'>
      <Avatar />
      <Avatar />
    </AvatarGroup>,
    'test-class',
  )

  it('should render group when multiple avatars are passed', () => {
    const { container, rerender } = renderWithNexUIProvider(
      <AvatarGroup>
        <Avatar />
      </AvatarGroup>,
    )

    expect(container.firstElementChild).toHaveClass(avatarClasses.root)

    rerender(
      <AvatarGroup>
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    expect(container.firstElementChild).toHaveClass(avatarGroupClasses.root)
  })

  it("should forward ref to AvatarGroup's root element", () => {
    const ref = createRef<HTMLDivElement>()
    const { getByTestId } = renderWithNexUIProvider(
      <AvatarGroup ref={ref} data-testid='avatar-group'>
        <Avatar />
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const root = getByTestId('avatar-group')
    expect(root).toBe(ref.current)
  })

  it('should forward classes to surplus slot', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <AvatarGroup
        data-testid='avatar-group'
        classes={{ surplus: 'test-class' }}
        max={2}
      >
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const root = getByTestId('avatar-group')
    expect(root.lastElementChild).toHaveClass('test-class')
  })

  it('should render `+N` when there are more than 4 avatars', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <AvatarGroup data-testid='avatar-group' max={4}>
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const root = getByTestId('avatar-group')
    expect(root.childNodes.length).toBe(5)
    expect(root.lastElementChild?.textContent).toBe('+4')
  })

  it('should render `+N` when total exceeds max children', () => {
    const { getByTestId, rerender, container } = renderWithNexUIProvider(
      <AvatarGroup data-testid='avatar-group' max={3} total={5}>
        <Avatar />
      </AvatarGroup>,
    )

    const root = getByTestId('avatar-group')
    expect(root.childNodes.length).toBe(2)
    expect(root.lastElementChild?.textContent).toBe('+4')

    rerender(
      <AvatarGroup max={3} total={10}>
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const updatedRoot = container.firstElementChild
    expect(updatedRoot?.childNodes.length).toBe(4)
    expect(updatedRoot?.lastElementChild?.textContent).toBe('+7')
  })

  it('should render custom surplus content', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <AvatarGroup
        max={5}
        data-testid='avatar-group'
        renderSurplus={() => <Avatar>More</Avatar>}
      >
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const root = getByTestId('avatar-group')
    expect(root.lastElementChild?.textContent).toBe('More')
  })

  it('should customize spacing between avatars', () => {
    const { container } = renderWithNexUIProvider(
      <AvatarGroup spacing={10}>
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const root = container.firstElementChild
    expect(root).toHaveStyle({ '--avatar-group-spacing': '10px' })
  })

  it('avatar should extend appearance styles', () => {
    const { container } = renderWithNexUIProvider(
      <AvatarGroup size='lg' color='blue' outlined radius='full'>
        <Avatar />
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const avatar = container.firstElementChild?.firstElementChild
    expect(avatar).toHaveClass(avatarClasses['size-lg'])
    expect(avatar).toHaveClass(avatarClasses['color-blue'])
    expect(avatar).toHaveClass(avatarClasses.outlined)
    expect(avatar).toHaveClass(avatarClasses['radius-full'])
  })

  it('should forward slotProps to surplus slot', () => {
    const { container } = renderWithNexUIProvider(
      <AvatarGroup slotProps={{ surplus: { className: 'test-class' } }} max={2}>
        <Avatar />
        <Avatar />
        <Avatar />
      </AvatarGroup>,
    )

    const surplusAvatar = container.firstElementChild?.lastElementChild
    expect(surplusAvatar).toHaveClass('test-class')
  })
})
