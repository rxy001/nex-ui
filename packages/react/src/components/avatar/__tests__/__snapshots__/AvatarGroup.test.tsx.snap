// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup should render with default props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--nui-colors-gray-contrast-text);
  background-color: var(--nui-colors-gray-primary);
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: var(--nui-sizes-10);
  height: var(--nui-sizes-10);
  font-size: var(--nui-font-sizes-lg);
  border-radius: var(--nui-radii-lg);
  -webkit-transition: var(--nui-transitions-transform);
  transition: var(--nui-transitions-transform);
  position: relative;
  z-index: 1;
}

.emotion-1 svg {
  font-size: 1.6em;
}

.emotion-1:not(:first-of-type) {
  -webkit-margin-start: var(--avatar-group-spacing, calc(var(--nui-spaces-2) * -1));
  margin-inline-start: var(--avatar-group-spacing, calc(var(--nui-spaces-2) * -1));
}

.emotion-1:not(:last-of-type):is(:hover, [data-hover=true]):not(:disabled, [data-disabled=true]) {
  -webkit-transform: translateX(min(0px, var(--avatar-group-spacing, calc(var(--nui-spaces-2) * -1))));
  -moz-transform: translateX(min(0px, var(--avatar-group-spacing, calc(var(--nui-spaces-2) * -1))));
  -ms-transform: translateX(min(0px, var(--avatar-group-spacing, calc(var(--nui-spaces-2) * -1))));
  transform: translateX(min(0px, var(--avatar-group-spacing, calc(var(--nui-spaces-2) * -1))));
}

<div
  class="nui-avatar-group-root emotion-0"
>
  <div
    class="nui-avatar-root emotion-1"
    data-color="gray"
    data-in-group="true"
    data-outlined="false"
    data-radius="md"
    data-size="md"
    data-testid="avatar"
  />
  <div
    class="nui-avatar-root emotion-1"
    data-color="gray"
    data-in-group="true"
    data-outlined="false"
    data-radius="md"
    data-size="md"
  />
  <div
    class="nui-avatar-root emotion-1"
    data-color="gray"
    data-in-group="true"
    data-outlined="false"
    data-radius="md"
    data-size="md"
  />
</div>
`;
